From fe3bc437d5b38f9705c5ba76a0199cd3b53a415b Mon Sep 17 00:00:00 2001
Message-Id: <fe3bc437d5b38f9705c5ba76a0199cd3b53a415b.1433460787.git.chang-joon.lee@intel.com>
From: Praveen Paneri <praveen.paneri@intel.com>
Date: Thu, 14 May 2015 11:18:21 +0530
Subject: [PATCH] SQUASHME! [VPG]: drm/i915: Avoid disabling Image Enhancement
 bit in DPST disable path

Although DPST is only supported on pipes A & B an issue with the DPST Core logic
is applying pipe B's DPST correction to pipe C. Thus, to avoid visual artifacts
on pipe C when pipe B is turned off we can't disable Image Enhancement bit.

SQUASHME! - This patch should be squashed into the following existing patch:
commit e5b6c85f3ba1f167695b17ef7857d34a6cb66fc6
Change-Id: I24147618fd125abed61b09f140644d5735f2c386
Author: Daisy Sun <daisy.sun@intel.com>
Date:   Fri Jan 3 16:59:05 2014 -0800
  MUST_REBASE [VPG]: drm/i915 Enable DPST on Haswell

Issue: GMINL-9406
Change-Id: I2cd1a903d696acde59c23209f7cb7df9d0d472e8
Signed-off-by: Praveen Paneri <praveen.paneri@intel.com>
---
 drivers/gpu/drm/i915/i915_dpst.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_dpst.c b/drivers/gpu/drm/i915/i915_dpst.c
index a29f830..5c114d5 100644
--- a/drivers/gpu/drm/i915/i915_dpst.c
+++ b/drivers/gpu/drm/i915/i915_dpst.c
@@ -146,7 +146,6 @@ i915_dpst_disable_hist_interrupt(struct drm_device *dev)
 	/* Disable histogram logic */
 	blm_hist_ctl = I915_READ(dev_priv->dpst.reg.blm_hist_ctl);
 	blm_hist_ctl &= ~IE_HISTOGRAM_ENABLE;
-	blm_hist_ctl &= ~(dev_priv->dpst.ie_mod_table_enable);
 
 	I915_WRITE(dev_priv->dpst.reg.blm_hist_ctl, blm_hist_ctl);
 
-- 
1.9.1

